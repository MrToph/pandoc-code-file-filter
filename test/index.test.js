const { execSync } = require("child_process");

function execPandocOnFile(fileName) {
  const stdout = execSync(
    `pandoc -s -t markdown test/examples/${fileName} --filter bin/filter.js`
  );
  return String(stdout);
}

test("line ranges", () => {
  const output = execPandocOnFile(`lines.md`);
  expect(output).toMatchSnapshot();
});

test("regex sections", () => {
  const output = execPandocOnFile(`regex.md`);
  expect(output).toMatchSnapshot();
});

test("section offsets", () => {
  const output = execPandocOnFile(`offset.md`);
  expect(output).toMatchSnapshot();
});

test("no dedent", () => {
  const output = execPandocOnFile(`no-dedent.md`);
  expect(output).toMatchSnapshot();
});
