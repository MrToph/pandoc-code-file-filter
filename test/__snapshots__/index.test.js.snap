// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`line ranges 1`] = `
"\`\`\` {.javascript}
// match line numbers 232-5353
// or regexp /asdf/ - /wasd/
const sectionRegexp = /^\\\\s*(\\\\d*|\\\\/.*\\\\/)\\\\s*-\\\\s*(\\\\d*|\\\\/.*\\\\/)\\\\s*$/i

const isRegExp = marker => marker.startsWith(\`/\`)

// NOT IN FILE: some explanation for the next section
if(isRegExp(end)) {
    endLineNumber = lines.length - 1;
} else if (end === \`\`) {
    // can leave out end marker
    endLineNumber = lines.length - 1;
} else {
    endLineNumber = parseInt(end, 10);
}
\`\`\`
"
`;

exports[`no dedent 1`] = `
"\`\`\` {.javascript}
    const matches = sectionDefinition.match(sectionRegEx)

    // empty section definition or text that does not match a section definition is just returned the same way
    if(!matches) return sectionDefinition
\`\`\`
"
`;

exports[`regex sections 1`] = `
"\`\`\` {.javascript}
const extractSectionFactory = lines => sectionDefinition => {
    const matches = sectionDefinition.match(sectionRegEx)

    // empty section definition or text that does not match a section definition is just returned the same way
    if(!matches) return sectionDefinition

    const [_, begin, end] = matches;
    console.warn(begin, end);

    let beginLineNumber = null;
    let endLineNumber = null;

    if(isRegExp(begin)) {

    } else {
        beginLineNumber = parseInt(begin, 10);
    }

    if(isRegExp(end)) {
        endLineNumber = lines.length - 1;
    } else if (end === \`\`) {
        // can leave out end marker
        endLineNumber = lines.length - 1;
    } else {
        endLineNumber = parseInt(end, 10);
    }

    return lines.slice(beginLineNumber - 1, endLineNumber).join(\`\\\\n\`)
\`\`\`
"
`;

exports[`section offsets 1`] = `
"\`\`\` {.javascript}
const extractSectionFactory = lines => sectionDefinition => {
    const matches = sectionDefinition.match(sectionRegEx)

    // empty section definition or text that does not match a section definition is just returned the same way
    if(!matches) return sectionDefinition

    const [_, begin, end] = matches;
    console.warn(begin, end);

    let beginLineNumber = null;
    let endLineNumber = null;

    if(isRegExp(begin)) {

    } else {
        beginLineNumber = parseInt(begin, 10);
    }

    if(isRegExp(end)) {
        endLineNumber = lines.length - 1;
    } else if (end === \`\`) {
        // can leave out end marker
        endLineNumber = lines.length - 1;
    } else {
        endLineNumber = parseInt(end, 10);
    }

    return lines.slice(beginLineNumber - 1, endLineNumber).join(\`\\\\n\`)
}

function xor(a,b) {
    return a ^ b
}
\`\`\`
"
`;

exports[`works with (escaped) slashes in regex 1`] = `
"\`\`\` {.javascript}
// match line numbers 232-5353
// or regexp /asdf/ - /wasd/
const sectionRegexp = /^\\\\s*(\\\\d*|\\\\/.*\\\\/)\\\\s*-\\\\s*(\\\\d*|\\\\/.*\\\\/)\\\\s*$/i

const isRegExp = marker => marker.startsWith(\`/\`)

const extractSectionFactory = lines => sectionDefinition => {
    const matches = sectionDefinition.match(sectionRegEx)

    // empty section definition or text that does not match a section definition is just returned the same way
\`\`\`
"
`;
